packages/vue-components/src/Submenu.vue

```diff
@@ -0,0 +1,183 @@
+<template>
+  <li
+    ref="submenu"
+    :class="[addClass, 'dropdown-submenu',
+             { 'dropright': dropright, 'dropleft': dropleft }]"
+  >
+    <slot name="button">
+      <a
+        class="submenu-toggle"
+        role="button"
+        :class="{disabled: disabled}"
+      >
+        <slot name="_header">
+          <slot name="header"></slot>
+        </slot>
+      </a>
+    </slot>
+    <slot name="dropdown-menu">
+      <ul class="dropdown-menu">
+        <slot></slot>
+      </ul>
+    </slot>
+  </li>
+</template>
+
+<script>
+import { toBoolean } from './utils/utils';
+import $ from './utils/NodeList';
+import positionSubmenu from './utils/submenu';
+
+export default {
+  props: {
+    addClass: {
+      type: String,
+      default: '',
+    },
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  data() {
+    return {
+      show: false,
+      dropright: true,
+      dropleft: false,
+    };
+  },
+  computed: {
+    disabledBool() {
+      return toBoolean(this.disabled);
+    },
+    showBool() {
+      return toBoolean(this.show);
+    },
+    slots() {
+      return this.$slots.default;
+    },
+  },
+  methods: {
+    hideSubmenu() {
+      this.show = false;
+      $(this.$refs.submenu).findChildren('ul').each(ul => ul.classList.toggle('show', false));
+      this.alignMenuRight();
+    },
+    showSubmenu() {
+      this.show = true;
+      $(this.$refs.submenu).findChildren('ul').each((ul) => {
+        ul.classList.toggle('show', true);
+        if (positionSubmenu.isRightAlign(ul)) {
+          this.alignMenuRight();
+        } else {
+          this.alignMenuLeft();
+        }
+        positionSubmenu.preventOverflow(ul);
+      });
+    },
+    alignMenuRight() {
+      this.dropright = true;
+      this.dropleft = false;
+    },
+    alignMenuLeft() {
+      this.dropright = false;
+      this.dropleft = true;
+    },
+  },
+  mounted() {
+    const $el = $(this.$refs.submenu);
+    if (this.show) {
+      this.showSubmenu();
+    }
+    $el.onBlur(() => { this.hideSubmenu(); }, false);
+    $el.findChildren('a,button').on('click', (e) => {
+      e.preventDefault();
+      if (e.target !== e.currentTarget) { e.stopPropagation(); }
+      if (window.innerWidth < 768) {
+        if (this.disabledBool) { return false; }
+        if (this.showBool) {
+          this.hideSubmenu();
+        } else {
+          this.showSubmenu();
+        }
+      }
+      return false;
+    });
+    $el.findChildren('a,button').on('mouseover', (e) => {
+      e.preventDefault();
+      if (window.innerWidth > 767) {
+        if (this.disabledBool) { return false; }
+        e.currentTarget.click();
+        this.showSubmenu();
+      }
+      return false;
+    });
+  },
+  beforeDestroy() {
+    const $el = $(this.$refs.submenu);
+    $el.offBlur();
+    $el.findChildren('a,button').off();
+    $el.findChildren('ul').off();
+  },
+};
+</script>
+
+<style scoped>
+.dropdown-submenu {
```

Comment:
let's indent these to fit the stylelint rules too (they've never been validated in our scripts for `.vue` files but should be)

Reply:
Updated :)